### PR TITLE
update tour/implicit-conversions for Scala 3

### PR DESCRIPTION
**Pull request by bishabosha**
adapt Tour of Scala implicit conversions page for scala 3

### CODE REVIEW

File: _tour/implicit-conversions.md [hunk 1]
```diff
@@ -10,46 +10,33 @@ previous-page: implicit-parameters
 redirect_from: "/tutorials/tour/implicit-conversions.html"
 ---
 
-An implicit conversion from type `S` to type `T` is defined by an implicit value which has function type `S => T`, or by an implicit method convertible to a value of that type.
+{% tabs implicit-conversion-defn class=tabs-scala-version %}
+{% tab 'Scala 2' %}
+In Scala 2, an implicit conversion from type `S` to type `T` is defined by an [implicit value]({% link _tour/implicit-parameters.md %}) which has function type `S => T`, or by an implicit method convertible to a value of that type.
+{% endtab %}
+{% tab 'Scala 3' %}
+In Scala 3, an implicit conversion from type `S` to type `T` is defined by a [given instance]({% link _tour/implicit-parameters.md %}) which has type `scala.Conversion[S, T]`, or by an implicit method which can be eta-expanded to the function type `S => T`.
```
Comment (julienrf) [inline _tour/implicit-conversions.md:18]: I think in Scala 3 the support for implicit methods (and implicit values of type functions) is there only for compatibility with Scala 2, no? I would rather say that in Scala 3 an implicit conversion is defined by a `given` instance of type `Conversion`. And then, say that for compatibility with Scala 2, they can also be defined as `implicit` methods (and tell the readers that they will find more information on the Scala 2 tab).

File: _tour/implicit-conversions.md [hunk 2]
```diff
@@ -10,46 +10,33 @@ previous-page: implicit-parameters
...
+An example is to pass a `scala.Int`, e.g. `x`, to a method that expects `scala.Long`. In this case, the implicit conversion `Int.int2long(x)` is inserted.
 
-The implicitly imported object `scala.Predef` declares several aliases to frequently used types (e.g. `scala.collection.immutable.Map` is aliased to `Map`) and methods (e.g. `assert`) but also several implicit conversions.
 
-For example, when calling a Java method that expects a `java.lang.Integer`, you are free to pass it a `scala.Int` instead. That's because Predef includes the following implicit conversions:
+In the second case, a conversion `c` is searched for which is applicable to `e` and whose result contains a member named `m`.
 
-```scala mdoc
-import scala.language.implicitConversions
+An example is to compare two strings `"foo" < "bar"`. In this case, `String` has no member `<`, so the implicit conversion `Predef.augmentString("foo") < "bar"` is inserted.
```
Comment (julienrf) [inline _tour/implicit-conversions.md:34]: Maybe remind that `Predef.*` is always imported by Scala programs (by default).

File: _tour/implicit-conversions.md [hunk 3]
```diff
@@ -58,3 +45,17 @@ To turn off the warnings take either of these actions:
 * Invoke the compiler with `-language:implicitConversions`
 
 No warning is emitted when the conversion is applied by the compiler.
+{% endtab %}
+{% tab 'Scala 3' %}
+Because implicit conversions can have pitfalls if used indiscriminately the compiler warns in two situations:
+- when compiling the implicit conversion definition (for Scala 2 style conversions).
```
Comment (julienrf) [inline _tour/implicit-conversions.md:51]: If the Scala 3 compiler does not warn on `given Conversion` instances, then I would make that more explicit:

```suggestion
- when compiling a Scala 2 style implicit conversion definition.
```

**Review by julienrf — CHANGES_REQUESTED** [review]
Thank you Jamie. I would expect also an explanation of where does the compiler search conversions. It can be a link to the implicit-parameters page maybe?

File: _tour/implicit-conversions.md (hunk 1 above)
Comment (bishabosha) [reply _tour/implicit-conversions.md:18]: that's a good suggestion, I will follow up with that

File: _tour/implicit-conversions.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ redirect_from: "/tutorials/tour/implicit-conversions.html"
 In Scala 2, an implicit conversion from type `S` to type `T` is defined by an [implicit value]({% link _tour/implicit-parameters.md %}) which has function type `S => T`, or by an implicit method convertible to a value of that type.
 {% endtab %}
 {% tab 'Scala 3' %}
-In Scala 3, an implicit conversion from type `S` to type `T` is defined by a [given instance]({% link _tour/implicit-parameters.md %}) which has type `scala.Conversion[S, T]`, or by an implicit method which can be eta-expanded to the function type `S => T`.
+In Scala 3, an implicit conversion from type `S` to type `T` is defined by a [given instance]({% link _tour/implicit-parameters.md %}) which has type `scala.Conversion[S, T]`. For compatibility with Scala 2, they can also be defined by an implicit method (read more in the Scala2 tab).
```
Comment (julienrf) [inline _tour/implicit-conversions.md]:
```suggestion
In Scala 3, an implicit conversion from type `S` to type `T` is defined by a [given instance]({% link _tour/implicit-parameters.md %}) which has type `scala.Conversion[S, T]`. For compatibility with Scala 2, they can also be defined by an implicit method (read more in the Scala 2 tab).
```

Comment (julienrf) [reply _tour/implicit-conversions.md]: You already changed it

**Pull request merged**